core/common/src/main/java/alluxio/underfs/UnderFileStatus.java

```diff
@@ -11,26 +11,101 @@
 
 package alluxio.underfs;
 
+import alluxio.Constants;
+
+import com.google.common.base.Objects;
+
 import javax.annotation.concurrent.NotThreadSafe;
 
 /**
  * Information about a file or a directory in the under file system. Listing contents in a
  * {@link UnderFileSystem} returns entries of this class.
  */
 @NotThreadSafe
-public class UnderFileStatus {
-  private final boolean mIsDirectory;
-  private final String mName;
+public final class UnderFileStatus {
+  /** Size of a directory. */
+  public static final long INVALID_CONTENT_LENGTH = 0L;
+
+  /** Last modified time for a directory. */
+  public static final long INVALID_MODIFIED_TIME = 0L;
+
+  private long mContentLength;
+  private boolean mIsDirectory;
+  private long mLastModifiedTimeMs;
+  private String mName;
+
+  // Permissions
+  private String mOwner;
+  private String mGroup;
+  private short mMode;
```

Comment:
@calvinjia The issue right now is that `UnderFileStatus` stores relative paths for listStatus which requires change the field `name`. Ideally this should be fully qualified path with scheme and stored in a UFSUri and not a string. I suggest we leave this for a later PR.